native/src/routes/feedback/containers/FeedbackModalContainer.js

```diff
@@ -1,33 +1,296 @@
+  offers?: Array<OfferModel>
+}
+
+type ContainerPropsType = {|
+  navigation: NavigationScreenProp<{|...NavigationLeafRoute, params: FeedbackInformationType |}>,
+  dispatch: Dispatch<StoreActionType>,
+  cities: $ReadOnlyArray<CityModel>,
+  t: TFunction
+|}
+
+type OwnPropsType = {|
+  navigation: NavigationScreenProp<{|...NavigationLeafRoute, params: FeedbackInformationType |}>,
+  t: TFunction
+|}
+
+type RefreshPropsType = {|
+  navigation: NavigationScreenProp<*>
+|}
+
+type StatePropsType = StatusPropsType<ContainerPropsType, RefreshPropsType>
+type DispatchPropsType = {| dispatch: Dispatch<StoreActionType> |}
+type PropsType = {| ...OwnPropsType, ...StatePropsType, ...DispatchPropsType |}
+
+const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsType => {
+  const refreshProps = {
+    navigation: ownProps.navigation
+  }
+  if (state.cities.status === 'error') {
+    return { status: 'error', message: state.cities.message, code: state.cities.code, refreshProps }
+  }
+
+  if (state.cities.status === 'loading') {
+    return { status: 'loading', progress: 0 }
+  }
+  return {
+    status: 'success',
+    innerProps: {
+      cities: state.cities.models,
+      navigation: ownProps.navigation,
+      t: ownProps.t
+    },
+    refreshProps
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>): DispatchPropsType => ({ dispatch })
+
+const refresh = (refreshProps: RefreshPropsType, dispatch: Dispatch<StoreActionType>) => {
+  const { navigation } = refreshProps
+  const feedbackInformation = navigation.state.params
+  const navigateToFeedback = createNavigateToFeedbackModal(navigation)
+  navigateToFeedback(feedbackInformation)
+}
+
+export type SendingStatusType = 'idle' | 'sending' | 'failed' | 'successful'
+
+type FeedbackModalStateType = {|
+  feedbackOptions: Array<FeedbackVariant>,
+  selectedFeedbackIndex: number,
+  comment: string,
+  sendingStatus: SendingStatusType
+|}
+
+class FeedbackModalContainer extends React.Component<ContainerPropsType, FeedbackModalStateType> {
+  constructor (props: ContainerPropsType) {
+    super(props)
+    const feedbackOptions = this.getFeedbackOptions()
+    this.state = { feedbackOptions, selectedFeedbackIndex: 0, comment: '', sendingStatus: 'idle' }
+  }
+
+  getCityName = (): string => {
+    const { cities, navigation } = this.props
+    const cityCode = navigation.getParam('cityCode')
+    return cityCode ? CityModel.findCityName(cities, cityCode) : INTEGREAT_INSTANCE
+  }
+
+  getFeedbackOptions = (): Array<FeedbackVariant> => {
+    const options = []
+    const currentPageFeedbackOption = this.getCurrentPageFeedbackOption()
+    if (currentPageFeedbackOption) {
+      options.push(currentPageFeedbackOption)
+    }
+
+    const contentFeedbackOption = this.getContentFeedbackOption()
+    if (contentFeedbackOption) {
+      options.push(contentFeedbackOption)
+    }
+
+    this.getOffersFeedbackOptions().forEach(option => options.push(option))
```

Comment:
I think push also works for multiple options and empty arrays: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push.
```suggestion
    options.push(...this.getOffersFeedbackOptions())
```